transaction.go

```diff
@@ -281,22 +286,24 @@ func threadedInitTransaction(t *Transaction) {
 	t.status = txnStatusWritten
 
 	// write the metadata and first page
-	buf := make([]byte, pageSize)
+	buf := bufPool.Get().([]byte)
+	defer bufPool.Put(buf)
+	for i := range buf {
+		buf[i] = 0
+	}
 	binary.LittleEndian.PutUint64(buf[:], t.status)
 	binary.LittleEndian.PutUint64(buf[8:], t.sequenceNumber)
 	var c checksum // checksum is left blank until transaction is committed
 	n := copy(buf[16:], c[:])
 	binary.LittleEndian.PutUint64(buf[n+16:], t.firstPage.nextOffset())
-	binary.LittleEndian.PutUint64(buf[n+24:], uint64(len(t.firstPage.payload)))
-	copy(buf[n+32:], t.firstPage.payload)
+	copy(buf[n+24:], t.firstPage.payload)
 	if _, err := t.wal.logFile.WriteAt(buf, int64(t.firstPage.offset)); err != nil {
 		t.initErr = errors.Extend(err, errors.New("Writing the first page to disk failed"))
 		return
 	}
 	// write subsequent pages
 	for page := t.firstPage.nextPage; page != nil; page = page.nextPage {
-		b := page.appendTo(buf[:0])
-		if _, err := t.wal.logFile.WriteAt(b, int64(page.offset)); err != nil {
+		if err := t.writePage(page); err != nil {
```

Comment:
`t.writePage` calls `bufPool.Get()` itself without overwriting the buffer with zeros. Is that ok? Seems like it might cause problems.

Reply:
I'm not sure what you mean. `t.writePage` clears the buffer.

Reply:
Oh sry, you are right. I missed that you added that.